php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1045 @@
+            ),
+            f.declareStatementWithOneStatement(
+                DECLARE_STATEMENT_HEAD(),
+                STATEMENT()
+            ),
+            f.alternativeDeclareStatement(
+                DECLARE_STATEMENT_HEAD(),
+                b.token(COLON),
+                b.zeroOrMore(INNER_STATEMENT()),
+                b.token(PHPKeyword.ENDDECLARE),
+                EOS()
+            )
+        ));
+  }
+
+  public DeclareStatementHead DECLARE_STATEMENT_HEAD() {
+    return b.<DeclareStatementHead>nonterminal()
+        .is(f.declareStatementHead(
+            b.token(PHPKeyword.DECLARE),
+            b.token(LPARENTHESIS),
+            MEMBER_CONST_DECLARATION(),
+            b.zeroOrMore(f.newTuple20(b.token(COMMA), MEMBER_CONST_DECLARATION())),
+            b.token(RPARENTHESIS)
+        ));
+  }
+
+  public InlineHTMLTree INLINE_HTML() {
+    return b.<InlineHTMLTree>nonterminal(PHPLexicalGrammar.INLINE_HTML_STATEMENT)
+        .is(f.inlineHTML(b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  public StatementTree INNER_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.INNER_STATEMENT)
+        .is(b.firstOf(
+            FUNCTION_DECLARATION(),
+//            CLASS_DECLARATION(),
+//            INTERFACE_DECLARATION(),
+            STATEMENT()
+        ));
+  }
+
+
+  public GlobalStatementTree GLOBAL_STATEMENT() {
+    return b.<GlobalStatementTree>nonterminal(PHPLexicalGrammar.GLOBAL_STATEMENT)
+        .is(f.globalStatement(
+            b.token(PHPKeyword.GLOBAL),
+            // fixme (Lena) : should be GLOBAL_VAR instead of COMPOUND_VARIABLE
+            COMPOUND_VARIABLE(),
+            b.zeroOrMore(f.newTuple16(b.token(COMMA), COMPOUND_VARIABLE())),
+            EOS()
+        ));
+  }
+
+  public UnsetVariableStatementTree UNSET_VARIABLE_STATEMENT() {
+    return b.<UnsetVariableStatementTree>nonterminal(PHPLexicalGrammar.UNSET_VARIABLE_STATEMENT)
+        .is(f.unsetVariableStatement(
+            b.token(PHPKeyword.UNSET),
+            b.token(LPARENTHESIS),
+            MEMBER_EXPRESSION(),
+            b.zeroOrMore(f.newTuple14(b.token(COMMA), MEMBER_EXPRESSION())),
+            b.token(RPARENTHESIS),
+            EOS()
+        ));
+  }
+
+  public YieldStatementTree YIELD_STATEMENT() {
+    return b.<YieldStatementTree>nonterminal(PHPLexicalGrammar.YIELD_STATEMENT)
+        .is(f.yieldStatement(YIELD_EXPRESSION(), EOS()));
+  }
+
+  public SwitchStatementTree SWITCH_STATEMENT() {
+    return b.<SwitchStatementTree>nonterminal(PHPLexicalGrammar.SWITCH_STATEMENT)
+        .is(b.firstOf(
+            f.switchStatement(
+                b.token(PHPKeyword.SWITCH),
+                PARENTHESIZED_EXPRESSION(),
+                b.token(PHPPunctuator.LCURLYBRACE),
+                b.optional(b.token(PHPPunctuator.SEMICOLON)),
+                b.zeroOrMore(SWITCH_CASE_CLAUSE()),
+                b.token(PHPPunctuator.RCURLYBRACE)
+            ),
+            f.alternativeSwitchStatement(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)